Sources/TSCUtility/Netrc.swift

```diff
@@ -0,0 +1,166 @@
+import Foundation
+import TSCBasic
+
+
+/// Supplies `Authorization` header, typically to be appended to `URLRequest`
+public protocol AuthorizationProviding {
+    /// Optional `Authorization` header, likely added to `URLRequest`
+    func authorization(for url: Foundation.URL) -> String?
+}
+
+extension AuthorizationProviding {
+    public func authorization(for url: Foundation.URL) -> String? {
+        return nil
+    }
+}
+
+@available (OSX 10.13, *)
+/// Container of parsed netrc connection settings
+public struct Netrc: AuthorizationProviding {
+    
+    /// Representation of `machine` connection settings & `default` connection settings.  If `default` connection settings present, they will be last element.
```

Comment:
**nitpick:** Please avoid long lines as these require horizontal scrolling when reading this code. I think the recommended line length is about 100-120